routes/recent.go

```diff
@@ -103,8 +114,20 @@ func Recent(w http.ResponseWriter, r *http.Request) {
 
 		r.Hole = hole.ByID[holeID]
 		r.Lang = lang.ByID[langID]
-
-		recents = append(recents, r)
+		r.Scoring = strings.Title(r.Scoring)
+
+		// If all of the information in two rows matches, other than the scoring, collapse them into one.
+		if previous.Login == r.Login &&
+			previous.Hole.ID == r.Hole.ID &&
+			previous.Lang.ID == r.Lang.ID &&
+			previous.Strokes == r.Strokes &&
+			(previous.Rank == r.Rank || previous.Rank > 3 && r.Rank > 3) &&
+			(r.Rank > 3 || (previous.TieCount > 0) == (r.TieCount > 0)) {
+			recents[len(recents)-1].Scoring = "Both"
+		} else {
+			recents = append(recents, r)
+			previous = r
+		}
```

Comment:
ah i see the grouping is done at the application level rather than the sql level, I guess this means the row count could fluctuate from 25-50 depending on how many rows are collapsed. Probably not the end of the world and I presume getting the SQL to apple the merging before the limit is harder and probably fairly expensive.

Reply:
Yeah I didn't really want to think about how to do it in SQL but I'm sure it's possible. Perhaps we should double the number of rows shown so that the amount of data people see doesn't suddenly decrease?